master/internal/rm/kubernetesrm/job.go

```diff
@@ -0,0 +1,751 @@
+	rmevents.Publish(j.allocationID, &sproto.ReleaseResources{Reason: "priority changed"})
+}
+
+func (j *job) changePosition() {
+	j.syslog.Info("interrupting job to change positions")
+	rmevents.Publish(j.allocationID, &sproto.ReleaseResources{Reason: "queue position changed"})
+}
+
+func (j *job) Kill() {
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	j.syslog.Info("received request to stop job")
+	if j.jobExitCause == nil {
+		j.jobExitCause = &exitReason{msg: "externally issued kill"}
+	}
+	j.kill()
+}
+
+func (j *job) kill() {
+	if !j.resourcesDeleted.CompareAndSwap(false, true) {
+		return
+	}
+
+	j.syslog.Infof("requesting to delete kubernetes resources %s", j.jobName)
+	j.resourceRequestQueue.deleteKubernetesResources(j.namespace, j.jobName, j.configMapName, "")
+}
+
+func (j *job) killPod(name string) {
+	if j.podKillSent[name] {
+		return
+	}
+
+	j.syslog.Infof("requesting to delete kubernetes resources %s", j.jobName)
+	j.resourceRequestQueue.deleteKubernetesResources(j.namespace, "", "", name)
+	j.podKillSent[name] = true
+}
+
+func (j *job) getNodeInfoForPods() []podNodeInfo {
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	var infos []podNodeInfo
+	for _, nodeName := range j.podNodeNames {
+		infos = append(infos, podNodeInfo{
+			nodeName:  nodeName,
+			numSlots:  j.slotsPerPod,
+			slotType:  j.slotType,
+			container: j.container.DeepCopy(),
+		})
+	}
+	return infos
+}
+
+func (j *job) startPodLogStreamers() error {
+	podList, err := j.podInterface.List(context.TODO(), metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", determinedLabel, j.req.AllocationID),
+	})
+	if err != nil {
+		return fmt.Errorf("listing job pods to reattach log streamers: %w", err)
+	}
+	for _, pod := range podList.Items {
+		if pod.Status.Phase != k8sV1.PodRunning {
+			j.syslog.Warnf("skipped reattaching pod log streamer for pod %s in phase %s", pod.Name, pod.Status.Phase)
+			continue
+		}
+
+		err := startPodLogStreamer(j.podInterface, pod.Name, func(log []byte) {
+			j.receiveContainerLog(sproto.ContainerLog{
+				Timestamp: time.Now().UTC(),
+				RunMessage: &aproto.RunMessage{
+					Value:   string(log),
+					StdType: stdcopy.Stdout,
+				},
+			})
+		})
+		if err != nil {
+			return fmt.Errorf("starting pod logs streamer for %s: %w", pod.Name, err)
+		}
+	}
+	return nil
+}
+
+func (j *job) createSpecAndSubmit() error {
+	jobSpec, configMapSpec, err := j.createSpec(j.scheduler)
+	if err != nil {
+		return err
+	}
+
+	j.resourceRequestQueue.createKubernetesResources(jobSpec, configMapSpec)
+	return nil
+}
+
+func (j *job) receiveResourceCreationFailed(msg resourceCreationFailed) {
+	j.syslog.WithError(msg.err).Error("pod handler notified that resource creation failed")
+	j.insertLog(time.Now().UTC(), msg.err.Error())
+}
+
+func (j *job) receiveResourceCreationCancelled() {
+	j.syslog.Info("pod creation canceled")
+	j.resourcesDeleted.Store(true)
+}
+
+func (j *job) receiveResourceDeletionFailed(err resourceDeletionFailed) {
```

Comment:
can we name the input variable in (err resourceDeletionFailed) as `msg` like in earlier functions? avoids an unfortunate looking `err.err`